ecoscope/analysis/UD/etd_range.py

```diff
@@ -83,7 +83,7 @@ class Weibull3Parameter(WeibullPDF):
 
 
 def calculate_etd_range(
-    trajectory_gdf: Trajectory,
+    trajectory_gdf: gpd.GeoDataFrame,
```

Comment:
Another way of putting this, if this can only operate on Trajectories, than it should be a method (i.e. accessor) on a Trajectory. (Which is the accessors paradigm, is a GeoDataFrame that adheres to a validated schema.)

Reply:
```python

traj = ...  # but, how do we get this? 🤷 

traj.calculate_etd()

```

Reply:
```python

plain_gdf = ...

traj = plain_gdf.ecoscope.EcoTrajectory()
turn_angle = traj.ecotrajectory.get_turn_angle()
etd = traj.ecotrajectory.calculate_etd()
```

🤔

Reply:
(Typing for alex...)

```python
plain_gdf.EcoscopeBase.init()
inited_gdf.relocations.thing
```

Reply:
```python
from typing import TypeVar

import pandera as pa
import pandas as pd

S = TypeVar("S")


class RelocationsSchema(pa.Schema): ...

class TrajectorySchema(pa.Schema): ...

class GDFWithSchema(Generic[S]):  ...


@pd.api.extensions.register_dataframe_accessor("ecoscope")
class ecoscope():

    def __init__(self, pandas_obj):
        assert isinstance(obj, gpd.GeoDataFrame)
        self._gdf = pandas_obj

    def Relocations(...) -> GDFWithSchema[RelocationsSchema]: 
        ...

    def Trajectory(...) -> GDFWithSchema[TrajectorySchema]:
        ...
        

```

Reply:
```python
plain_gdf = ...

reloc = plain_gdf.ecoscope.Relocations()
```